### DISCUSSION TITLE
Failed to install ruby 2.6.10_1! on M2 Mac

### DISCUSSION THREAD

**Discussion by Derek-Schutt** [Homebrew/discussions#4523] [Getting started]
### Output of `brew config`

```shell
$ brew config
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:61029cec31c68a1fae1fa90fa876adf43d0becff777da793f9b5c5577f00567a
Already downloaded: /Users/derek/Library/Caches/Homebrew/portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
==> Pouring portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
Error: Failed to install ruby 2.6.10_1!
Error: Failed to upgrade Homebrew Portable Ruby!
```

### Output of `brew doctor`

```shell
$ brew doctor
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:61029cec31c68a1fae1fa90fa876adf43d0becff777da793f9b5c5577f00567a
Already downloaded: /Users/derek/Library/Caches/Homebrew/portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
==> Pouring portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
Error: Failed to install ruby 2.6.10_1!
Error: Failed to upgrade Homebrew Portable Ruby!
```

### Description of issue

I have just done a fresh install of homebrew on a new M2 Macbook Pro running Ventura 13.2.1.    Although the installation is shown as successful, whenever I try to do something I get "Failed to install ruby 2.6.10_1!".    I've tried uninstalling and reinstalling homebrew, but that doesn't seem to work.   Can anyone help here?

**Comment by SMillerDev** [comment]
Can you try running those commands with `--debug`?

**Comment by SMillerDev** [comment]
Can you try running those commands with `--debug`?

**Reply by Derek-Schutt** [reply]
It shows the same thing...

$ brew doctor --debug
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:61029cec31c68a1fae1fa90fa876adf43d0becff777da793f9b5c5577f00567a
Already downloaded: /Users/derek/Library/Caches/Homebrew/portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
==> Pouring portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
Error: Failed to install ruby 2.6.10_1!
Error: Failed to upgrade Homebrew Portable Ruby!

**Comment by oceanminded28** [comment]
Same issue for me on a new MacBook Pro

MacBook-Pro:~ jlevesque$ brew install ruby@2.6
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:61029cec31c68a1fae1fa90fa876adf43d0becff777da793f9b5c5577f00567a
Already downloaded: /Users/jlevesque/Library/Caches/Homebrew/portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
==> Pouring portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
Error: Failed to install ruby 2.6.10_1!
Error: Failed to upgrade Homebrew Portable Ruby!

**Reply by gromgit** [reply]
Try removing `/Users/jlevesque/Library/Caches/Homebrew/portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz` and doing the install again.

**Reply by oceanminded28** [reply]
I have and get the same error. I have tried removing Xcode and reinstalling and various other methods but to no avail. I have seen incompatibility with Xcode 14 but all of that is from 6mos or more ago so I am not sure that should be the issue. I can't run any brew commands without the error. `brew update-reset` was also tried

**Reply by oceanminded28** [reply]
Running Ventura 13.4 and Xcode 14.3. Also this is a fresh computer I am setting up

**Reply by oceanminded28** [reply]
`MacBook-Pro:~ jlevesque$ rm /Users/jlevesque/Library/Caches/Homebrew/portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
MacBook-Pro:~ jlevesque$ brew doctor
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:61029cec31c68a1fae1fa90fa876adf43d0becff777da793f9b5c5577f00567a
######################################################################### 100.0%
==> Pouring portable-ruby-2.6.10_1.el_capitan.bottle.tar.gz
Error: Failed to install ruby 2.6.10_1!
Error: Failed to upgrade Homebrew Portable Ruby!`

**Reply by oceanminded28** [reply]
Output from `ruby -v`
`ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.arm64e-darwin22]`

**Comment by Derek-Schutt** [comment]
I got things to work by installing rosetta 2.   Now homebrew works fine.

**Reply by oceanminded28** [reply]
Thank you!! How is this not better documented somewhere? In any case I am all good now thank you

**Reply by Bo98** [reply]
> How is this not better documented somewhere?

Rosetta should only be needed if installing into `/usr/local`, where Homebrew is run in x86_64 mode.

`/opt/homebrew` should be fully arm64 native.
